src/autopas/containers/verletClusterLists/VerletClusterCellsParticleIterator.h

```diff
@@ -0,0 +1,266 @@
+ * VerletClusterCellsRegionParticleIterator to iterate over all particles within a specific region
+ * @tparam Particle Particle type over which the iterator iterates
+ * @tparam ParticleCell Cell type over which the iterator iterates
+ */
+template <class Particle, class ParticleCell, bool modifiable>
+class VerletClusterCellsRegionParticleIterator
+    : public VerletClusterCellsParticleIterator<Particle, ParticleCell, modifiable> {
+  using CellVecType = std::conditional_t<modifiable, std::vector<ParticleCell>, const std::vector<ParticleCell>>;
+  using DummyStartsType = std::conditional_t<modifiable, std::vector<size_t> *, const std::vector<size_t>>;
+
+ public:
+  /**
+   * Constructor of the VerletClusterCellsRegionParticleIterator.
+   *
+   * @param cont Container of particle cells.
+   * @param dummyStarts indices of the first dummy particle for each cell
+   * @param startRegion Lower corner of the region to iterate over.
+   * @param endRegion Top corner of the region to iterate over.
+   * @param indicesInRegion List of indices all threads will iterate over.
+   * @param offsetToDummy offset to add to create dummy particle
+   * @param behavior The IteratorBehavior that specifies which type of cells shall be iterated through.
+   * @param skin skin to which extend serch window
+   */
+  explicit VerletClusterCellsRegionParticleIterator(CellVecType *cont, DummyStartsType dummyStarts,
+                                                    std::array<double, 3> startRegion, std::array<double, 3> endRegion,
+                                                    const std::vector<size_t> &indicesInRegion, double offsetToDummy,
+                                                    IteratorBehavior behavior = haloAndOwned, double skin = 0.0)
+      : VerletClusterCellsParticleIterator<Particle, ParticleCell, modifiable>(cont, dummyStarts, offsetToDummy,
+                                                                               behavior),
+        _startRegion(startRegion),
+        _endRegion(endRegion),
+        _indicesInRegion(indicesInRegion),
+        _currentRegionIndex(0),
+        _skin(skin) {
+    _indicesInRegion.push_back(this->_vectorOfCells->size() + 1);
+    this->cellId = _indicesInRegion[_currentRegionIndex];
+    this->cellIter = (*this->_vectorOfCells)[this->cellId]._particles.begin();
+    this->cellEnd = this->cellIter + this->getDummyStartbyIndex(this->cellId);
+    --this->cellIter;
+    VerletClusterCellsRegionParticleIterator<Particle, ParticleCell, modifiable>::operator++();
+  }
+
+  /**
+   * @copydoc ParticleIteratorInterface::operator++()
+   */
+  inline VerletClusterCellsParticleIterator<Particle, ParticleCell, modifiable> &operator++() override {
+    do {
+      ++this->cellIter;
+
+      while (this->cellIter == this->cellEnd) {
+        this->cellId = _indicesInRegion[++_currentRegionIndex];
+        if (not this->isValid()) {
+          return *this;
+        }
+        this->cellIter = std::lower_bound(
```

Comment:
add a comment, that this also is valid if particles are a tiny bit unsorted (by skin/2)